easybuild/easyblocks/generic/intelbase.py

```diff
@@ -131,10 +131,9 @@ def parse_components_list(self):
                                      % self.cfg['components'])
         else:
             self.install_components = []
-            for comp in available_components:
-                for comp_regex in self.cfg['components']:
-                    if re.search(comp_regex, comp):
-                        self.install_components.append(comp)
+            for comp_regex in self.cfg['components']:
+                comps = [comp for comp in available_components if re.match(comp_regex, comp)]
```

Comment:
OK, as discussed, it's more about the list of component regexes is going to be interpreted (by humans).

what does 'comp' mean? only *exactly* comp (no with `search`, yes with `match`)?
or everything with `comp` in it, including stuff I may not be aware of (yes, with `search`, no with `match`)

if you want everything with `comp` in it, enforcing the use of `.*comp.*` is crystal clear imho